### PR TITLE
test(styles): updates for v10

### PR DESCRIPTION
**Pull request by asudoh**
#### Changelog

**Changed**

- Updates to tests so flipping feature flags won't break them.
  - Added explicit feature flags for tests that are valid only with certain set of feature flags.
  - Filtering out warnings of non-default feature flag usages.
  - Disabled "pure modules" tests for removed files.

#### Testing / Reviewing

Need to ensure CI passes with or without feature flags flipped.

### CODE REVIEW

**Comment by asudoh** [comment]
@shinytoyrobots @alisonjoseph Test-only change - Good to put in v10? Thanks!

**Comment by joshblack** [comment]
@asudoh @alisonjoseph would we want to stick with the work in #2208 for this already? I think the `renderClassic` pattern would be helpful to reduce churn here, also don't mind having the `warn` being called twice, added comments in the relevant files saying why they should be called twice too 👍

**Comment by asudoh** [comment]
@joshblack Updated this PR that incorporates your test code from #2208 - Does it look good for you? Thanks!

**Comment by asudoh** [comment]
@alisonjoseph @joshblack TL/DR - Still need test change in addition to what we have in #2208, details below:

Updated Karma test/config so test passes with feature flags flipped (i.e. tests with `v10` markup). Some highlights are:

1. File uploader test needed an update to cope with a change between `v9` and `v10` wrt where `bx--file-close` class is set to (`<svg>` vs. `<button>`)
2. Still need a update to data table (likely in `src`) - Update: Created #2220.
3. Karma configuration change to exclude removed files in `v10`, (though @joshblack's `v10` change (#2208) effectively covers this item)

File: src/globals/grid/__tests__/grid-test.js [hunk 1]
```diff
@@ -8,12 +8,38 @@
...
+const renderClassic = content =>
+  render(`
+$feature-flags: (components-x: false, breaking-changes-x: false, grid: false);
+${content}
+`);
+
+const isClassic = async () => {
+  const { calls } = await renderClassic(`
+@import '../../scss/functions';
+$t: test(feature-flag-enabled('braking-changes-x'));
```
Comment (emyarod) [inline src/globals/grid/__tests__/grid-test.js]:
```suggestion
$t: test(feature-flag-enabled('breaking-changes-x'));
```

is this a typo? appears a few times

File: src/globals/scss/__tests__/css--plex-core-test.js [hunk 2]
```diff
@@ -7,14 +7,29 @@
...
+  render(`
+$css--reset: false;
+$feature-flags: (components-x: false, breaking-changes-x: false);
+${content}
+`);
+
+const isClassic = async () => {
+  const { calls } = await renderClassic(`
+@import '../../scss/functions';
+$t: test(feature-flag-enabled('braking-changes-x'));
```
Comment (emyarod) [inline src/globals/scss/__tests__/css--plex-core-test.js]:
```suggestion
$t: test(feature-flag-enabled('breaking-changes-x'));
```

File: src/globals/scss/__tests__/typography-test.js [hunk 3]
```diff
@@ -19,20 +19,35 @@ const variables = [
...
+const renderClassic = content =>
+  render(`
+$feature-flags: (components-x: false, breaking-changes-x: false);
+${content}
+`);
+
+const isClassic = async () => {
+  const { calls } = await renderClassic(`
+@import '../../scss/functions';
+$t: test(feature-flag-enabled('braking-changes-x'));
```
Comment (emyarod) [inline src/globals/scss/__tests__/typography-test.js]:
```suggestion
$t: test(feature-flag-enabled('breaking-changes-x'));
```

**Review by emyarod — APPROVED** [review]
looks good to me assuming we clear up the overlap with #2208! just left a few comments about the feature flag name

**Review by joshblack — APPROVED** [review]

**Pull request merged**